docs/best-practices.md

```diff
@@ -0,0 +1,49 @@
+### How to select good environmental variable name?
+When selecting for environmental variable name, please check Dotnet cli [convention examples](https://github.com/dotnet/sdk/blob/56a6239ba8959df4e7d851923e5022a0fce78805/src/Cli/dotnet/Program.cs#L158-L165).
+
+Please note below is not ultimate guide, just our learning from last experience and hoping to help you and it can change depending on circumstance so feel free to improve it.
+
+* Use capitalized snake case for the environment variable name (ex. CAPITALIZED_SNAKE_CASE). Because env var name is case sensitive on Linux.
+* Separate words by underscore(`_`) to for readibility and clarity.
+* Value doesn't have to case sensitive, we can convert to [uppercase](https://github.com/NuGet/NuGet.Client/blob/680f9bd4e97db7cd7482584276886764de69d3cb/src/NuGet.Core/NuGet.Packaging/PackageArchiveReader.cs#L530) and compare to validate value.
```

Comment:
If we're writing a guide, then we should be explicit about what should/should not be done. 

Consider rewording this to: 

* Value should be [case insensitive](link)